src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -563,14 +559,14 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         current_item: &Option<String>,
         extra_fragment: &Option<String>,
     ) -> Option<Res> {
-        let check_full_res_inner = |this: &Self, result: Result<Res, ErrorKind<'_>>| {
+        let check_full_res_inner = |this: &Self, result: Result<Res, ErrorKind>| {
```

Comment:
This function could be reworked to not have any closures and not be longer as it is currently. I wanted to do it, but for the sake of minimalism, this PR is a mess right now anyway and doesn't need any more.

Reply:
Sure, seems like a good follow-up but I agree it should be separate.